homeassistant/components/stream/__init__.py

```diff
@@ -272,11 +276,11 @@ def add_provider(
         if not (provider := self._outputs.get(fmt)):
 
             @callback
-            def idle_callback() -> None:
+            def idle_callback(hass: HomeAssistant) -> None:
                 if (
                     not self.keepalive or fmt == RECORDER_PROVIDER
                 ) and fmt in self._outputs:
-                    self.remove_provider(self._outputs[fmt])
+                    hass.async_create_task(self.remove_provider(self._outputs[fmt]))
```

Comment:
I believe check idle needs to have the output removed so this removal cannot be deferred

Reply:
I'm not sure if that implies the idle timer must also be async or if the stop call in remove should be the task instead, or it returns a future for the caller to decide after it does the removal.

Reply:
Can probably just make the callback async and call it from fire